server.py

```diff
@@ -1,58 +1,78 @@
-import threading
-from flask import Flask
-from flask_socketio import SocketIO, emit
-from canvas_board import CanvasBoard, CanvasBoardEncoder
+import random, string
+from flask import Flask, jsonify
+from flask_socketio import SocketIO, emit, join_room, leave_room
+from canvas_board import CanvasBoard
+from json import dumps
 
 # constants
 PORT = 8080
 HEIGHT = 500
 WIDTH = 500
+ID_LENGTH = 7
 
 # create Flask object
-app = Flask(__name__)
+app = Flask(__name__, template_folder='static/', static_folder='static/')
 app.config['SECRET_KEY'] = 'secret!'
 socketio = SocketIO(app, cors_allowed_origins='*')
 
 # server class
 class Server:
-    def __init__(self, imagedata):
-        self.board = CanvasBoard(imagedata)
-        self.lock = threading.Lock()
-
-    def update_board(self, diffs: list):
-        self.lock.acquire()
-        self.board.update_board(diffs)
-        self.lock.release()
+    def __init__(self, boards: 'dict that maps room id to CanvasBoard'):
+        self.boards = boards
 
+    def update_board(self, diffs: list, room_id: str):
+        self.boards[room_id].update_board(diffs)
 
 # instantiate server class for board state
 imagedata = {
     'width': WIDTH,
     'height': HEIGHT,
     'data': [0 for i in range(4 * WIDTH * HEIGHT)]
 }
-server = Server(imagedata)
+server = Server({})
 
 @app.route('/')
 def index():
     return app.send_static_file('index.html')
 
+@app.route('/create')
+def create():
+    room = None
+    while room in server.boards or room is None:
+        room = ''.join(random.choice(string.ascii_uppercase + string.digits) for _ in range(ID_LENGTH))
+        print(room)
+    response = jsonify(room_id=room)
+    # just return response with room id
+    # don't worry about frontend
+    return response
+
 @socketio.on('connect', namespace='/canvas')
 def connect_canvas():
-    # broadcast board upon initial connect at /canvas endpoint
-    # turn board into JSON
-    board = CanvasBoardEncoder().encode(server.board)
-    emit('initialize-board', board)
+    print("connected to websocket")
 
 @socketio.on('disconnect')
 def on_disconnect():
     print("disconnected from websocket")
 
 @socketio.on('send-stroke', namespace="/canvas")
-def handle_send_stroke(stroke):
-    server.update_board(stroke['diffs'])
-    # broadcast the new change
-    emit('broadcast-stroke', stroke, broadcast=True)
+def handle_send_stroke(stroke, room_data):

```

Comment:
What do you think about standardizing the parameter of socket.io to just be a single `payload` arg which will simply be a JSON containing appropriate fields?  In this case it would be:

```
payload = {
  diffs: [
    {"coord": 0, "val": 0}
  ],
  room_id: "some_id"
}
```

WDYT?